query-service/api.raml

```diff
@@ -2,293 +2,400 @@
             type: error.ServerError
 
-# TODO: need the file upload patch in the gradle plugin for this
-/word-count:
   post:
-    description: Counts the words in an uploaded file.
+    displayName: Create New Job
+    description: Creates a new blast query job.
     body:
       multipart/form-data:
-        type: file
+        type: object
+        additionalProperties: false
+        properties:
+          config:
+            displayName: Job Configuration
+            type: lib.QueryJobRequest
+            required: true
+          query:
+            displayName: Query File
+            description: |
+              Optional file upload parameter for uploading multi-blast queries.
+              
+              **IMPORTANT**: Either this field or the job config's `query` field
+              MUST be set in order for a job to be valid.
+            type: file
+            required: false
     responses:
       200:
         description: |
-          Success
-          
-          Job was processed and queued successfully.
+          Query job was successfully submitted to the job queue for execution.
+        body:
+          application/json:
+            type: lib.QueryJobDetails
+      400:
+        description: |
+          Invalid request body or parameters.
+        body:
+          application/json:
+            type: error.BadRequestError
+      401:
+        description: |
+          No valid user authentication provided.
+        body:
+          application/json:
+            type: error.UnauthorizedError
+      422:
+        description: |
+          Invalid job configuration.
+        body:
+          application/json:
+            type: error.UnprocessableEntityError
+
+
+# ---------------------------------------------------------------------------- #
+#                                                                              #
+#   Job By ID                                                                  #
+#                                                                              #
+# ---------------------------------------------------------------------------- #
+/jobs/{job-id}:
+  displayName: Job By ID
+  description: |
+    Operations on a singular job identified by the `{job-id}` URI parameter.
+
+  uriParameters:
+    job-id:
+      type: string
+      pattern: ^[\dA-Fa-f]{32}$
+      minLength: 32
+      maxLength: 32
+
+  get:
+    displayName: Get Job Details
+    description: |
+      Returns details about the target job.
+    
+      As a side effect, interactions with this endpoint will cause the target
+      job to be added to the requesting user's job collection if it was not
+      already present in that user's collection.  This default behavior can be
+      defeated by using the `skip_job_add` query parameter.
+    queryParameters:
+      skip_job_add:
+        displayName: Skip Job Adding
+        description: |
+          A flag, that when set to true, tells the service to skip the part of
```

Comment:
I wonder if skipping should be default here?  Then have property be addToUserCollection=true to override.  This would make it clear what's happening during sharing and I feel like this name implies idempotency (no one would think it would be added twice), so not confusing if already in collection.

Reply:
This is done to mimic the current, naive sharing approach.

Your suggestion is similar to what I want, which is basically just to make it so sharing is **NOT** the default behavior and has to be manually specified, but it was deemed that changing the behavior would be too much client work to implement.